packages/app/src/offscreen/Offscreen.ts

```diff
@@ -0,0 +1,54 @@
+import { RPCAction } from "@cryptkeeperzk/providers";
+import { FullProof, generateProof } from "@cryptkeeperzk/semaphore-proof";
+import { RequestHandler, SemaphoreProofRequest } from "@cryptkeeperzk/types";
+import { ZkIdentitySemaphore, getMerkleProof } from "@cryptkeeperzk/zk";
+
+import Handler from "@src/background/controllers/handler";
+import { Runtime } from "webextension-polyfill";
+
+const RPC_METHOD_ACCESS: Record<RPCAction, boolean> = {
+  ...Object.values(RPCAction).reduce((acc, method) => ({ ...acc, [method]: false }), {} as Record<RPCAction, boolean>),
+  [RPCAction.GENERATE_SEMAPHORE_PROOF_OFFSCREEN]: true,
+};
+
+export class OffscreenController {
```

Comment:
Why do you need another handler?

Reply:
It's needed to handle Offscreen RPC related calls.